bin/packages/build.js

```diff
@@ -114,6 +114,18 @@ function createStyleEntryTransform() {
 			const entries = await glob(
 				path.resolve( PACKAGES_DIR, packageName, 'src/*.scss' )
 			);
+
+			// Account for the specific case where block styles in
+			// block-library package also need rebuilding.
+			if (
+				packageName === 'block-library' &&
+				[ 'style.scss', 'editor.scss' ].includes(
```

Comment:
Maybe also add `theme.scss` here?
```suggestion
				[ 'style.scss', 'editor.scss', 'theme.scss' ].includes(
```

Reply:
@aristath Should that also be in the glob here?
https://github.com/WordPress/gutenberg/blob/a09da7a59ce61fb84aff2258bac19fb506776104/bin/packages/build.js#L195-L203

That's what I based the suggestion of style.scss and editor.scss on.

Reply:
Hmmm I could be mistake, but I think these are all compiled to separate files... Adding it there is part of another PR on https://github.com/WordPress/gutenberg/pull/31239. `theme.scss` files currently don't get compiled to a separate file for each block and instead they get combined to a single `theme.css` file for all blocks.

Reply:
OK. So just the first suggestion then, right?

Reply:
I feel like that means the suggestion shouldn't be applied. It'd result in a theme.scss file in the block build-style folder, which the full build doesn't do.

Reply:
Yep, it does result in a theme.css in the block's folder.

@aristath I think you'll need to apply the suggestion as part of #31239. This code maps the stylesheet entrypoints so should match what the full build does in the snippet I linked above (though it's only handling a single file).

Compiling the block editor package's full stylesheet already works as per:
https://wordpress.slack.com/archives/C02QB2JS7/p1619599534383600

Reply:
@talldan I'll add it to the other PR (#31239) so we'll get it when that gets merged :wink: 
At that point, we'll _need_ the separate `theme.css` files per-block.

Reply:
Great, sounds good 👍